test: `e2e_note_getter` not using `DocsExampleContract`

benesjan opened the pull request:
This PR is a piece of a series of PRs in which I clean up our use of test contracts. In this case I am replacing the use of DocsExampleContract in `e2e_note_getter.test.ts` with the goal of eventually not using `DocsExampleContract` in `/yarn-project` at all.

I introduce a new NoteGetter test contract.

benesjan committed:
test: NoteGetter test contract

benesjan commented:
* **#13366** <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/13366?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/13366?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `master`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

benesjan committed:
WIP

benesjan committed:
cleanup

benesjan committed:
comment

nventuro reviewed (approved):
Thanks! This one seems like a good example of the kind of tests that seem fine to be moved to TXE eventually - we're not really testing any e2e properties here.

benesjan committed:
stale issue cleanup

benesjan committed:
refactor: streamline note retrieval in NoteGetter contract

Updated the read_note_values function to directly initialize NoteViewerOptions within the view_notes call, improving code clarity and reducing unnecessary variable declarations.

benesjan commented on yarn-project/end-to-end/src/e2e_note_getter.test.ts:
```diff
@@ -46,106 +41,31 @@ describe('e2e_note_getter', () => {
       await contract.methods.insert_notes([0, 1, 2]).send().wait();
```
Tackled that in https://github.com/AztecProtocol/aztec-packages/pull/13366/commits/bfa4d6a603dde5dc4990ab4940675813557ed79b and closed the issue

benesjan commented on noir-projects/noir-contracts/contracts/test/note_getter_contract/src/main.nr:
```diff
@@ -0,0 +1,56 @@
+    #[utility]
+    unconstrained fn read_note_values(comparator: u8, value: Field) -> BoundedVec<Field, 10> {
+        let mut options = NoteViewerOptions::new();
+        let notes = storage.set.view_notes(options.select(
```
It worked https://github.com/AztecProtocol/aztec-packages/pull/13366/commits/21823fb00e830430ae539e3d58ddd771e7b3a4bb

benesjan committed:
Merge branch 'master' into 04-07-test_notegetter_test_contract
